fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1002,8 +1005,8 @@ def _set_is_root(self) -> None:
             return
         # No FullyShardedDataParallel instance wraps this, else _is_root would be set to False.
         self._is_root = True
-        assert self._queue_wait_for_post_backward_closure is None
-        self._queue_wait_for_post_backward_closure = self._queue_wait_for_post_backward
+        # Do not allow to checkpoint root instance and run forward multiple times.
```

Comment:
```suggestion
        # Check if the root instance is being checkpointed. It doesn't make sense to
        # checkpoint the root instance since it won't save GPU memory.
```